### ISSUE TITLE
Why is the FLOPs calculation multiplied by 2?

### ISSUE THREAD

**Issue by syz247179876** [ultralytics/ultralytics#19773]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

```python
def get_flops(model, imgsz=640):
  import thop
  import torch
  from copy import deepcopy
  p = next(model.parameters())
  if not isinstance(imgsz, list):
      imgsz = [imgsz, imgsz]  # expand if int/float
  try:
      # Use stride size for input tensor
      stride = max(int(model.stride.max()), 32) if hasattr(model, "stride") else 32  # max stride
      im = torch.empty((1, p.shape[1], stride, stride), device=p.device)  # input image in BCHW format
      flops = thop.profile(deepcopy(model), inputs=[im], verbose=False)[0] / 1e9 * 2  # stride GFLOPs
      return flops * imgsz[0] / stride * imgsz[1] / stride  # imgsz GFLOPs
  except Exception:
      # Use actual image size for input tensor (i.e. required for RTDETR models)
      im = torch.empty((1, p.shape[1], *imgsz), device=p.device)  # input image in BCHW format
      return thop.profile(deepcopy(model), inputs=[im], verbose=False)[0] / 1e9 * 2  # imgsz GFLOPs
```

I wonder why the FLOPs calculation here needs to be multiplied by 2.

Can I uniformly use this calculation method for all models?
`flops, params = thop.profile(model, inputs=(inputs,))` or `flops = fvcore.nn.FlopCountAnalysis(model, inputs)`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @syz247179876, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for detailed explanations and examples, which might help clarify your question. You can also explore our [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage sections for additional insights.

If this is a 🐛 Bug Report, kindly provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us diagnose the issue more effectively.

For ❓ Questions like this, please share as much context as possible, including any dependencies, relevant logs, or variations in your specific use case. It would also help to know if you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

💡 To ensure you're working with the latest updates and fixes, upgrade to the newest version of `ultralytics` in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO runs smoothly in several tested environments. Here are some options for you to try:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community Support

Join the Ultralytics community to connect with others and get assistance:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time conversations
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for knowledge sharing

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. These tests validate the correct operation of YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response to guide you in the right direction. An Ultralytics engineer will review your inquiry and provide further assistance soon. Thank you for your patience and support 🌟!

**Comment by glenn-jocher** [comment]
@syz247179876 the FLOPs calculation is multiplied by 2 because THOP counts Multiply-Add Operations (MACs), where each MAC consists of 2 floating point operations. This conversion from MACs to FLOPs ensures consistency with standard FLOPs reporting conventions. 

The shown method works for all Ultralytics-supported models except architectures like RT-DETR that require full-image input sizing. For implementation details, see the [model profiling source code](https://docs.ultralytics.com/reference/utils/torch_utils/#ultralytics.utils.torch_utils.get_flops).

**Comment by syz247179876** [comment]
@glenn-jocher Thank you for your explanation!

**Comment by glenn-jocher** [comment]
@syz247179876 you're welcome! For clarity, the ×2 multiplier converts THOP's reported MACs (Multiply-Add Operations) to FLOPs (1 MAC = 2 FLOPs). The current implementation handles most architectures well except special cases like RT-DETR requiring full-image inputs. For implementation details, see the [model profiling source code](https://docs.ultralytics.com/reference/utils/torch_utils/#ultralytics.utils.torch_utils.get_flops). 😊
